airbyte-integrations/connectors/source-mailchimp/source_mailchimp/source.py

```diff
@@ -21,70 +21,31 @@
-
-from airbyte_protocol import (
-    AirbyteCatalog,
-    AirbyteConnectionStatus,
-    AirbyteMessage,
-    ConfiguredAirbyteCatalog,
-    ConfiguredAirbyteStream,
-    Status,
-    SyncMode,
-)
-from base_python import AirbyteLogger, Source
-
-from .client import Client
-
-
-class SourceMailchimp(Source):
-    """
-    Mailchimp API Reference: https://mailchimp.com/developer/api/
-    """
-
-    def check(self, logger: AirbyteLogger, config: json) -> AirbyteConnectionStatus:
-        client = self._client(config)
-        alive, error = client.health_check()
-        if not alive:
-            return AirbyteConnectionStatus(status=Status.FAILED, message=f"{error.title}: {error.detail}")
-
-        return AirbyteConnectionStatus(status=Status.SUCCEEDED)
-
-    def discover(self, logger: AirbyteLogger, config: json) -> AirbyteCatalog:
-        client = self._client(config)
-
-        return AirbyteCatalog(streams=client.get_streams())
-
-    def read(
-        self, logger: AirbyteLogger, config: json, catalog: ConfiguredAirbyteCatalog, state: Dict[str, any]
-    ) -> Generator[AirbyteMessage, None, None]:
-        client = self._client(config)
-
-        logger.info("Starting syncing mailchimp")
-        for configured_stream in catalog.streams:
-            yield from self._read_record(client=client, configured_stream=configured_stream, state=state)
-
-        logger.info("Finished syncing mailchimp")
-
-    def _client(self, config: json):
-        client = Client(username=config["username"], apikey=config["apikey"])
-
-        return client
-
-    @staticmethod
-    def _read_record(
-        client: Client, configured_stream: ConfiguredAirbyteStream, state: DefaultDict[str, any]
-    ) -> Generator[AirbyteMessage, None, None]:
-        entity_map = {
-            "Lists": client.lists,
-            "Campaigns": client.campaigns,
-        }
-
-        stream_name = configured_stream.stream.name
-
-        if configured_stream.sync_mode == SyncMode.full_refresh:
-            state.pop(stream_name, None)
-
-        for record in entity_map[stream_name](state=state):
-            yield record
+import base64
+from typing import Any, List, Mapping, Tuple
+
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from mailchimp3 import MailChimp
+
+from .streams import Campaigns, EmailActivity, Lists
+
+
+class SourceMailchimp(AbstractSource):
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
+        try:
+            client = MailChimp(mc_api=config["apikey"], mc_user=config["username"])
+            client.ping.get()
+            return True, None
+        except Exception as e:
+            return False, e
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        auth_string = f"anystring:{config['apikey']}".encode("utf8")
```

Comment:
I think there is some miscommunication, I mean could you please inherit your class from TokenAuthenticator and have this encoding logic in its constructor.